### PR TITLE
[TIMOB-25279] iOS: Resolve all Xcode warnings

### PR DESCRIPTION
**Pull request by hansemannn**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-25279

(No unit-tests applicable due to internal changes only)

### CODE REVIEW

**Review by vijaysingh-axway — APPROVED** [review]
CR passed.

**Comment by ewieberappc** [comment]
FR Passed. Used the SDK to build multiple, varied projects using Xcode 9. No related warnings witnessed.

File: iphone/Classes/TiThreading.h [hunk 1]
```diff
@@ -40,14 +40,14 @@
   }
 
 //TODO: Now that we have TiThreadPerform, we should optimize this out.
-#define ENSURE_UI_THREAD_WITH_OBJ(x, y, z)                                         \
-  if (![NSThread isMainThread]) {                                                  \
+#define ENSURE_UI_THREAD_WITH_OBJ(x, y, z)                                          \
```
Comment (sgtcoolguy) [inline iphone/Classes/TiThreading.h]: clang-format is complaining (and rightfully so) that the line continuations don't line up

File: iphone/Classes/TiPublicAPI.h [hunk 2]
```diff
@@ -52,7 +52,7 @@
   }
 
 #define TI_PUBLIC_METHOD_END_ARGS(methodName, returnType) \
-  if (![@"void" isEqualToString:@"" #returnType]) {       \
+  if (![@"void" isEqualToString:@"" #returnType]) {      \
```
Comment (sgtcoolguy) [inline iphone/Classes/TiPublicAPI.h]: clang-format is complaining (and rightfully so) that the line continuations don't line up

**Comment by hansemannn** [comment]
@sgtcoolguy Comments addressed.

**Pull request merged**